test/case/ietf_interfaces/remove_ip_address.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python3
+
+import infamy
+
+#Name of the interface to remove IPv4 addresses from:
+interface_name = "eth0"
+
+def print_ip_addresses(target):
+    running = target.get_config_dict("ietf-interfaces:interfaces")
+    interfaces = running.get("interfaces", {}).get("interface", [])
+
+    if isinstance(interfaces, dict):
+        interfaces = [interfaces]  # Convert single interface to list
+
+    for interface in interfaces:
+        name = interface["name"]
+
+        if "ipv4" in interface:
+            ipv4_addresses = interface["ipv4"].get("address", [])
+            for ipv4_address in ipv4_addresses:
+                address = ipv4_address.get("ip")
+                prefix_length = ipv4_address.get("prefix-length")
+                print(f"Interface: {name}, IPv4 Address: {address}/{prefix_length}")
+        else:
+            print(f"Interface: {name}, No IPv4 Address")
+
+        if "ipv6" in interface:
+            ipv6_addresses = interface["ipv6"].get("address", [])
+            for ipv6_address in ipv6_addresses:
+                address = ipv6_address.get("ip")
+                prefix_length = ipv6_address.get("prefix-length")
+                print(f"Interface: {name}, IPv6 Address: {address}/{prefix_length}")
+        else:
+            print(f"Interface: {name}, No IPv6 Address")
+
+def assert_ipv4_removed(target, interface_name):
+    running = target.get_config_dict("ietf-interfaces:interfaces")
+    interfaces = running.get("interfaces", {}).get("interface", [])
+
+    if isinstance(interfaces, dict):
+        interfaces = [interfaces]  # Convert single interface to list
+
+    for interface in interfaces:
+        name = interface["name"]
+        if name == interface_name:
+            assert "ipv4" not in interface, f"IPv4 address not removed from interface {interface_name}"
+            return
+    print(f"No interface with the name {interface_name}")
+         
+with infamy.Test() as test:
+    with test.step("Setup"):
+        env = infamy.Env(infamy.std_topology("1x1"))
+        target = env.attach("target", "mgmt")
+
+    with test.step("Get initial IP addresses"):
+        print_ip_addresses(target)
+
+    with test.step(f"Remove IPv4 addresses from {interface_name}"):
+
+        config = {
+            "interfaces": {
+                "@xmlns": "urn:ietf:params:xml:ns:yang:ietf-interfaces",
+                "interface": [{
+                    "name": f"{interface_name}",
+                    "type": "iana-if-type:ethernetCsmacd",
+                    "ipv4": {
+                        "@xmlns": "urn:ietf:params:xml:ns:yang:ietf-ip",
+                        "operation": "remove"
```

Comment:
Rather than specifying operations explicitly, and having to parse/mangle the edit data in `netconf.py`, I think we should leverage the builtin features in `libyang` for creating diffs. See comments in `netconf.py`.